translate/storage/test_php.py

```diff
@@ -325,7 +343,7 @@ def test_parsing_define_entries_with_quotes(self):
         """Parse define syntax for entries with quotes"""
         phpsource = """define('_SETTINGS_COOKIEPREFIX', 'Prefixo da "cookie"');
 define('_YOUR_USERNAME', 'O seu nome de usuario: "cookie"');
-define("_REGISTER", "Register <a href=\"register.php\">here</a>");"""
+define("_REGISTER", "Register <a href=\\"register.php\\">here</a>");"""
```

Comment:
This was correctly escaped before.

Reply:
No - that there is escaping for Python and then for PHP to have quotes inside quotes.

The original code produces:

```php
define("_REGISTER", "Register <a href="register.php">here</a>");
```
What is not valid PHP code.

Reply:
Then it looks like the problem is in the original code, not in the test.

Reply:
Yes, the original PHP parser is broken in many ways, in this case it happily parsed invalid PHP code.

Reply:
I still don't think this fix in the test is the right way to do this.

Reply:
Okay the current situation is:

- The test contains invalid PHP code.
- The current php parser happily parses that.

Apparently I'm not going to fix broken parser to thrown error on this when there is  #3697. I'm open to other ways to fix this but I really don't see how :-).

Reply:
I don't see how

```php
"Register <a href=\"register.php\">here</a>"
```

is invalid PHP code. If I am not wrong this renders to valid HTML:

```html
Register <a href="register.php">here</a>
```

Reply:
But here is not `\"register.php\"`, but only `"register.php"`!

```python
>>> print """define("_REGISTER", "Register <a href=\\"register.php\\">here</a>");"""
define("_REGISTER", "Register <a href=\"register.php\">here</a>");
>>> print """define("_REGISTER", "Register <a href=\"register.php\">here</a>");"""
define("_REGISTER", "Register <a href="register.php">here</a>");
```

Reply:
Alternatively (if you don't like double escaping), this could be addressed by using raw string:

```patch
diff --git a/translate/storage/test_php.py b/translate/storage/test_php.py
index fa1b11ca..45e99483 100644
--- a/translate/storage/test_php.py
+++ b/translate/storage/test_php.py
@@ -323,7 +323,7 @@ define( '_CM_POSTED','Enviado');"""
 
     def test_parsing_define_entries_with_quotes(self):
         """Parse define syntax for entries with quotes"""
-        phpsource = """define('_SETTINGS_COOKIEPREFIX', 'Prefixo da "cookie"');
+        phpsource = r"""define('_SETTINGS_COOKIEPREFIX', 'Prefixo da "cookie"');
 define('_YOUR_USERNAME', 'O seu nome de usuario: "cookie"');
 define("_REGISTER", "Register <a href=\"register.php\">here</a>");"""
         phpfile = self.phpparse(phpsource)
```

Reply:
Works for me.